pkg/session/session.go

```diff
@@ -2430,6 +2430,11 @@ func runStmt(ctx context.Context, se *session, s sqlexec.Statement) (rs sqlexec.
 				}
 			}
 		}
+		if err == nil && sessVars.TxnCtx.CouldRetry && !s.IsReadOnly(sessVars) {
```

Comment:
This checking condition is confusing, could you explain it and why put it here?
IMO the `checkStmtLimit` should be put to a unified place as a hook which all SQL statement processing would use.

Reply:
Nice catch , already updated, PTAL